Improve existing docs (#45)
SAP/cloud-sdk: docs/js/faq.mdx

---
id: frequently-asked-questions
title: Frequently asked questions
hide_title: false
hide_table_of_contents: false
sidebar_label: FAQ
description: You asked! We Answered! We collected here the most frequent question about SAP Cloud SDK for JavaScript.
keywords:
  - sap
  - cloud
  - sdk
  - cloud native
  - cloud sdk
  - sap cloud sdk
---

## Generic questions

### Is SAP Cloud SDK for JavaScript Open Source

Yes.
The SAP Cloud SDK for JavaScript is Open Source and available under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).
You can find the source code on [GitHub](https://github.com/SAP/cloud-sdk-js) and the modules on [npm](https://www.npmjs.com/search?q=%40sap-cloud-sdk).
The SAP Cloud SDK command-line interface resides in a separate [GitHub repository](https://github.com/sap/cloud-sdk-cli).

> ### What is Virtual Data Model?

Reviewer: ```suggestion
### What is the Virtual Data Model?
```

> In simple words, Virtual Data Model is a nick-name for pre-generated type-safe client libraries for allowlisted OData services of SAP [S/4HANA Cloud](https://www.sap.com/germany/products/s4hana-erp.html).

Reviewer: ```suggestion
In simple words, the Virtual Data Model is a nick-name for pre-generated type-safe client libraries for allowlisted OData services of SAP [S/4HANA Cloud](https://www.sap.com/germany/products/s4hana-erp.html).
```

These libraries are a part of SAP Cloud SDK for JavaScript but they are **not** Open Source and shipped under SAP Developer license.

> Libraries are updated with every release of S/4HANA Cloud and you can always find them on [npm](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm-*)

Reviewer: ```suggestion
They are updated with every release of SAP S/4HANA Cloud and you can always find them on [npm](https://www.npmjs.com/search?q=%40sap%2Fcloud-sdk-vdm-*)
```

### Can I contribute to the SAP Cloud SDK for JavaScript?

We are happy to accept and review your valuable contribution.
For details on contributing check our [contribution guide](https://github.com/SAP/cloud-sdk/blob/main/CONTRIBUTING.md)

### How often do you release a new SDK version?

We usually release [bi-weekly](https://github.com/SAP/cloud-sdk-js/releases).
All features that are in _Generally Available_ or _Beta_ state get into the next release.

Pregenerated type-safe client libraries are released quarterly to follow the SAP S/4HANA Cloud release cycle.

### Do you release hotfixes?

Yes, we do.
They usually have a higher `patch` number according to [semver](https://semver.org/), i.e: 1.24.1 instead of 1.24.0.
Check our [release notes](https://github.com/SAP/cloud-sdk-js/releases) for more details.

### Should I update with every release?

The general recommendation is _YES_.
This way you'll reduce the effort to keep up with the fast pace of cloud development.
We try to keep stable functionality consistent and explicitly notify about breaking changes.

> Be cautious about using features annotated as _Beta_ because their API can change with every release.

Reviewer: ```suggestion
Be cautious about using features annotated with _experimental_ because their API can change with every release.
```

### Which Node versions are supported by the SDK?

We support and recommend the [latest active LTS version of Node](https://nodejs.org/en/about/releases/).
You can try using a later Node version on your own risk.

### Can I use features annotated as Beta in production?

> We **do not recommend** using API that is marked unstable in productive code.

Reviewer: ```suggestion
We **do not recommend** using APIs that are marked unstable in productive code.
```

We do not guarantee any API compatibility for future updates and the features might be experimental.

> You can use these features to test new functionality, provide us feedback, and plan migration steps when _Beta_ features are releases as _General Availability_.

Reviewer: ```suggestion
You can use these features to test new functionality, provide feedback, and plan migration steps when _Beta_ features are released as _General Availability_.
```
